racket/collects/racket/contract/private/hash.rkt

```diff
@@ -198,7 +198,8 @@
   (define this-immutable (base-hash/c-immutable ctc))
   (λ (fuel)
     (define rnd (random fuel))
-    (define rnd-len (add1 (modulo (add1 rnd) fuel)))
+    ;; keep amount of random elements reasonable, no more than 4
+    (define rnd-len (add1 (modulo (add1 fuel) 4)))
```

Comment:
What is the reason to generate only 4 elements? If you worry about the size of the generated hash table, I think the approach used by `(listof _)` can be applied here to decide the size of the hash table (with a certain probability increase the size and recur):
https://github.com/racket/racket/blob/master/racket/collects/racket/contract/private/list.rkt#L30-#L34

Reply:
The idea is that it can use the geometric distribution. The wikipedia page on it at the beginning is confusing to me, but a bit later down it says this:

> Consider a sequence of trials, where each trial has only two possible outcomes (designated 
> failure and success). The probability of success is assumed to be the same for each trial. In 
> such a sequence of trials, the geometric distribution is useful to model the number of failures
> before the first success. The distribution gives the probability that there are zero failures
> before the first success, one failure before the first success, two failures before the first 
> success, and so on.

So you can just write a loop that works like that.